trust/token.c

```diff
@@ -272,11 +283,25 @@ loader_load_directory (p11_token *token,
 		return 0;
 	}
 
+	paths = p11_array_new (NULL);
+
 	while ((dp = readdir (dir)) != NULL) {
 		path = p11_path_build (directory, dp->d_name, NULL);
 		return_val_if_fail (path != NULL, -1);
 
-		ret = loader_load_if_file (token, path);
+		if (!p11_array_push (paths, path)) {
+			free (path);
```

Comment:
A simple `return_val_if_fail` will suffice as `p11_array_push` will only fail in case of OOM in which case the program dies anyway.